### ISSUE TITLE
Dumping Ranking instances

### ISSUE THREAD

**Issue by mpkato** [mpkato/interleaving#34]
I have been thinking about how to pass the interleaving results to a production environment.

Probably, an instance of InterleavingMethod can have function `dump_rankings(self, file)`, which dumps sampled rankings into a file. Rankings, their probability, and some additional information (e.g. teams or credits) must be stored in the dump file. It is probably convenient to assign a unique ID for each ranking by using `hash`.

The format of the dump file should be something readable by the other programs, such as json or xml.

A production system can load the dump file, show the rankings to users, observe user clicks, and record the number of impressions and rank of clicks for each ranking.

`InterleavingMethod.load_rankings(cls, file)` can then load the rankings from the dump file and somehow obtain the number of impressions and rank of clicks. `InterleavingMethod.evaluate(cls, ranking, clicks)` can be used for deciding the ranking preference based on the loaded information.

In summary, my proposal is to implement:
1. `InterleavingMethod.dump_rankings(self, file)` for dumping sampled rankings
2. `InterleavingMethod.load_rankings(cls, file)` for loading the dumped rankings
3. `Ranking.dumps(self)` and `Ranking.loads(self, s)`
4. The format of the dump file can be 

```
{ 
    ranking_id: {
        'probability': float,
        'ranking': {
            'ranking_list': list,
            'teams': { ranker_index: list }
        }
    }
}
```

**Comment by tmanabe** [comment]
Here are some questions:
a. Could you give us an example dump file?
b. When is `Ranking::dumps()` used?
c. Should the name of `ranking_list` be `document_ids` or `document_id_list`?

**Comment by tmanabe** [comment]
@mpkato Have you started implementing this?
